api/envoy/extensions/filters/http/ext_proc/v3alpha/ext_proc.proto

```diff
@@ -0,0 +1,145 @@
+//
+// The protocol itself is based on a bidirectional gRPC stream. Envoy will send the
+// server
+// :ref:`ProcessingRequest <envoy_v3_api_msg_service.ext_proc.v3alpha.ProcessingRequest>`
+// messages, and the server must reply with
+// :ref:`ProcessingResponse <envoy_v3_api_msg_service.ext_proc.v3alpha.ProcessingResponse>`.
+
+// [#next-free-field: 10]
+message ExternalProcessor {
+  // Configuration for the gRPC service that the filter will communicate with.
+  // The filter supports both the "Envoy" and "Google" gRPC clients.
+  config.core.v3.GrpcService grpc_service = 1;
+
+  // By default, if the gRPC stream cannot be established, or if it is closed
+  // prematurely with an error, the filter will fail. Specifically, if the
+  // response headers have not yet been delivered, then it will return a 500
+  // error downstream. If they have been delivered, then instead the HTTP stream to the
+  // downstream client will be reset.
+  // With this parameter set to true, however, then if the gRPC stream is prematurely closed
+  // or could not be opened, processing continues without error.
+  bool failure_mode_allow = 2;
+
+  // Specifies default options for how HTTP headers, trailers, and bodies are
+  // sent. See ProcessingMode for details.
+  ProcessingMode processing_mode = 3;
+
+  // If true, send each part of the HTTP request or response specified by ProcessingMode
+  // asynchronously -- in other words, send the message on the gRPC stream and then continue
+  // filter processing. If false, which is the default, suspend filter execution after
+  // each message is sent to the remote service and wait up to "message_timeout"
+  // for a reply.
+  bool async_mode = 4;
+
+  // For each property name, select a matching property value to attach to the
+  // request message. Property names are the same ones supported in the WebAssembly
+  // runtime.
+  repeated string request_properties = 5;
+
+  // For each property name, select a matching property value to attach to the
+  // response message.
+  repeated string response_properties = 6;
+
+  // Specifies the timeout for each individual message sent on the stream and
+  // when the filter is running in synchronous mode. Whenever
+  // the proxy sends a message on the stream that requires a response, it will
+  // reset this timer, and will stop processing and return an error (subject
+  // to the processing mode) if the timer expires. There is no timeout when
+  // the filter is running in asynchronous mode. Default is 200 ms.
+  google.protobuf.Duration message_timeout = 7;
+
+  // Specifies if the filter is enabled.
+  //
+  // If :ref:`runtime_key <envoy_api_field_config.core.v3.RuntimeFractionalPercent.runtime_key>` is specified,
+  // Envoy will lookup the runtime key to get the percentage of requests to filter.
+  //
+  // If this field is not specified, the filter will be enabled for all requests.
+  config.core.v3.RuntimeFractionalPercent filter_enabled = 8;
```

Comment:
Should something like this really be promoted to HTTP filter wrapper message @envoyproxy/api-shepherds? It seems like this is something that should be implemented once in HCM, rather than each filter.

Reply:
I'd of course be happy if this weren't something I have to copy from another filter. Let me know if you think I should leave this field out for now and plan to add it back later.

Reply:
I'd suggest eliding it for now and filing an issue for a generic HCM fractional percent filter enablement. We can always add it back here if needed while that issue is pending.

Reply:
Ack -- I will take it out of this PR.